hoomd/hpmc/pytest/test_shape_move.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+from collections.abc import Sequence, Mapping
+import math
+from numbers import Number
+import hoomd
+import hoomd.write
+import hoomd.hpmc
+import numpy as np
+import pytest
+from copy import deepcopy
+import coxeter
+from coxeter.shapes import ConvexPolyhedron
+from hoomd.logging import LoggerCategories
+from hoomd.conftest import logging_check
+
+_seed = 0
+
+
+def _equivalent_data_structures(reference, struct_2):
+    """Compare arbitrary data structures for equality.
+
+    ``reference`` is expected to be the reference data structure. Cannot handle
+    set like data structures.
+    """
+    if isinstance(reference, np.ndarray):
+        return np.allclose(reference, struct_2)
+    if isinstance(reference, Mapping):
+        # if the non-reference value does not have all the keys
+        # we don't check for the exact same keys, since some values may have
+        # defaults.
+        if set(reference.keys()) - set(struct_2.keys()):
+            return False
+        return all(
+            _equivalent_data_structures(reference[key], struct_2[key])
+            for key in reference)
+    if isinstance(reference, Sequence):
+        if len(reference) != len(struct_2):
+            return False
+        return all(
+            _equivalent_data_structures(value_1, value_2)
+            for value_1, value_2 in zip(reference, struct_2))
+    if isinstance(reference, Number):
+        return math.isclose(reference, struct_2, rel_tol=1e-4)
+
+
+ttf = coxeter.families.TruncatedTetrahedronFamily()
+
+
+class TruncatedTetrahedron(hoomd.hpmc.shape_move.Callback):
+
+    def __getitem__(self, trunc):
+        shape = ConvexPolyhedron(
+            ttf.get_shape(trunc).vertices /
+            (ttf.get_shape(trunc).volume**(1 / 3)))
+        return [v for v in shape.vertices]
+
+    def __call__(self, params):
+        verts = self.__getitem__(params[0])
+        args = {'vertices': verts, 'sweep_radius': 0.0, 'ignore_statistics': 0}
+        return hoomd.hpmc._hpmc.PolyhedronVertices(args)
```

Comment:
Why does this test call internal HPMC C++ data structures? Is this the intended API for users?

Reply:
That's how it was done originally. However, with the latest v3 API improvements, we can simply return a `dict` and construct the shape internally. I'll fix it.